scripts/us_census/acs5yr/subject_tables/subject_table_test.py

```diff
@@ -0,0 +1,145 @@
+        if '_cleaned.csv' in filename:
+            csv_path = os.path.join(testdata_path, filename)
+        elif '_output.mcf' in filename:
+            statvar_mcf_path = os.path.join(testdata_path, filename)
+        elif 'column_map.json' in filename:
+            column_map_path = os.path.join(testdata_path, filename)
+        elif filename.endswith('.zip'):
+            zip_path = os.path.join(testdata_path, filename)
+
+    return spec_path, csv_path, statvar_mcf_path, column_map_path, zip_path
+
+
+class TestSubjectTable(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # A list of all directories to be tested
+        cls.test_dirs = []
+        for table_dir in os.listdir(_SCRIPT_PATH):
+            table_dir_path = os.path.join(_SCRIPT_PATH, table_dir)
+            process_path = os.path.join(_SCRIPT_PATH, table_dir, 'process.py')
+
+            # Run test only if table_dir is not in _EXCLUDE_DIRS and process.py
+            # is not in table_dir and table_dir is a directory
+            if table_dir not in _EXCLUDE_DIRS and os.path.isdir(
+                    table_dir_path) and not os.path.isfile(process_path):
+                cls.test_dirs.append(table_dir_path)
+
+    def test_csv_mcf_column_map(self):
+
+        for table_dir_path in self.test_dirs:
+            with self.subTest(table_dir_path=table_dir_path):
+                spec_path, csv_path, statvar_mcf_path, column_map_path, \
+                    zip_path = get_paths(table_dir_path)
+
+                # Check if all required files exist
+                self.assertTrue(spec_path is not None)
+                self.assertTrue(csv_path is not None)
+                self.assertTrue(statvar_mcf_path is not None)
+                self.assertTrue(column_map_path is not None)
+                self.assertTrue(zip_path is not None)
+
+                # TODO: Look into speeding up this section of the code
+                with tempfile.TemporaryDirectory() as tmp_dir:
+                    # Run scripts/us_census/acs5yr/subject_tables/process.py
+                    process_arg = ['python']
+                    process_arg.append(os.path.join(_SCRIPT_PATH, 'process.py'))
+                    process_arg.append('--table_prefix=test')
+                    process_arg.append(f'--spec_path={spec_path}')
+                    process_arg.append(f'--input_path={zip_path}')
+                    process_arg.append(f'--output_dir={tmp_dir}')
+                    process_arg.append('--has_percent=True')
+                    process_arg.append('--debug=False')
+
+                    subprocess.run(process_arg, check=True)
+
+                    test_mcf_path = os.path.join(tmp_dir, 'test_output.mcf')
+                    test_column_map_path = os.path.join(tmp_dir,
+                                                        'column_map.json')
+                    test_csv_path = os.path.join(tmp_dir, 'test_cleaned.csv')
+
+                    # Test StatVar MCF
```

Comment:
Can this be a member function that takes the two files to be compared as arguments?

Reply:
Done. Have moved the reading of files to a separate function. Retained the asserts in the unittest function.

Reply:
Done.